src/java/fr/paris/lutece/portal/service/plugin/PluginService.java

```diff
@@ -93,6 +102,53 @@ public static void init( ) throws LuteceInitException
         loadPlugins( );
     }
 
+    private static final Map<String, PluginFile> _mapPluginsMeta = new HashMap<>();
+
+    /**
+     * Pre-loads name and version from plugins to make them available (with getPluginMeta())
+     * if needed before complete loading by init().
+     * 
+     * @throws LuteceInitException
+     */
+    public static void preloadMeta() throws LuteceInitException
+    {
+        File dirPlugin = new File(AppPathService.getPath(PATH_PLUGIN));
+
+        if (dirPlugin.exists())// it should
+        {
+            // all plugins
+            List<File> files = new ArrayList<>(Arrays.asList(dirPlugin.listFiles(new FileListFilter("", EXTENSION_FILE))));
+            // and the core 
+            files.add(new File(AppPathService.getPath(PATH_CONF, CORE_XML)));
+            try
+            {
+                DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+                for (File file : files)
+                {
+                    Document doc = builder.parse(file);
+                    PluginFile pluginFile = new PluginFile();
+                    pluginFile.setVersion(doc.getElementsByTagName("version").item(0).getTextContent());
+                    pluginFile.setName(doc.getElementsByTagName("name").item(0).getTextContent());
+                    _mapPluginsMeta.put(pluginFile.getName(), pluginFile);
```

Comment:
This map could maybe  be used in loadPlugins() method ?
(to avoid duplicates)

Reply:
Not really : 
The first map (_mapPluginsMeta) contains incomplete PluginFile (**not** Plugin) instances that only contain the minimal set of metadata. These PluginFile instances are parsed "by hand". At this point, Plugin instances cannot be yet instantiated.
The second map (populated with loadPlugins) cannot reuse the first one, since the second deals with complete Plugin instances, that themselves need complete PluginFile instances (fully loaded through a proper deserialization mechanism that has code dependencies that we cannot use when loading the first map).